### PR TITLE
Adding visibility of docs when API is made public from the restricted by roles

### PR DESCRIPTION
**Pull request by Meruja**
Fix wso2/product-apim/issues/7448

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java [hunk 1]
```diff
@@ -2282,6 +2281,23 @@ public void testUpdateAPI_InCreatedState() throws RegistryException, UserStoreEx
...
+        Mockito.when(APIUtil.getAPIDocPath(api.getId())).thenReturn(documentation.getFilePath());
+        APIProviderImplWrapper apiProviderImplWrapper = new APIProviderImplWrapper(apimgtDAO, null, null);
+        Resource docResource = Mockito.mock(Resource.class);
+        Mockito.when(docResource.getUUID()).thenReturn(documentation.getId());
+        Mockito.when(apiProviderImplWrapper.registry.get(documentation.getFilePath())).thenReturn(docResource);
+
+        GenericArtifact docArtifact = Mockito.mock(GenericArtifact.class);
+        Mockito.when(artifactManager.getGenericArtifact(documentation.getId())).thenReturn(docArtifact);
+        Mockito.when(APIUtil.getDocumentation(docArtifact)).thenReturn(documentation);
+        Mockito.when(docArtifact.getPath()).thenReturn("artifact/path");
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: Shall we make this a constant?

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java]: fixed with f49c9d3

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -3154,6 +3163,57 @@ public void addDocumentationContent(API api, String documentationName, String te
...
+                    String filePath = docFilePath.substring(startIndex, docFilePath.length());
+                    APIUtil.setResourcePermissions(api.getId().getProviderName(), api.getVisibility(),
+                            authorizedRoles, filePath, registry);
+                }
+            } catch (UserStoreException e) {
+                throw new APIManagementException("Error in retrieving Tenant Information while adding api :"
+                        + api.getId().getApiName(), e);
+            }
+        } catch (RegistryException e) {
+            handleException("Failed to update visibility of documentation", e);
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:3214]: Add context information to error logs

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 3]
```diff
@@ -3154,6 +3163,57 @@ public void addDocumentationContent(API api, String documentationName, String te
...
+                String docFilePath = artifact.getAttribute(APIConstants.DOC_FILE_PATH);
+                if (org.apache.commons.lang.StringUtils.isEmpty(docFilePath)) {
+                    int startIndex = docFilePath.indexOf("governance") + "governance".length();
+                    String filePath = docFilePath.substring(startIndex, docFilePath.length());
+                    APIUtil.setResourcePermissions(api.getId().getProviderName(), api.getVisibility(),
+                            authorizedRoles, filePath, registry);
+                }
+            } catch (UserStoreException e) {
+                throw new APIManagementException("Error in retrieving Tenant Information while adding api :"
+                        + api.getId().getApiName(), e);
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:3211]: This operation is not related to API creation right? This is related to doc visibility update. Please modify the error messages accrdingly.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:3214]: Fixed with ec30343

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 3 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:3211]: Fixed with ec30343
